### PR TITLE
CassandraConnection cleanup

### PR DESCRIPTION
**Pull request by conorbros**
### CassandraConnection changes
- Moved `Request` struct to be private to `CassandraConnection`. This means that the `Request` used for `ConnectionPool` doesn't need to have an `Option<i16>` field for the message id.
- added a `send` method. Transforms can just give the connection pool their message and return channel instead of having to interact with the implementation of `CassandraConnection`.

### Other changes
- Renamed `messages` in `Request` to `message` because it only holds one message.
- Changed `Response` from a tuple to a struct. This was used lots and it was confusing. Sometimes it was used without the `Response` type and instead another tuple was declared.

### CODE REVIEW

File: shotover-proxy/src/transforms/cassandra/connection.rs [hunk 1]
```diff
@@ -36,26 +45,45 @@ impl<C: Codec + 'static> CassandraConnection<C> {
...
         self.connection = Some(out_tx);
         Ok(())
     }
+
+    /// Send a `Message` to this `CassandraConnection` and expect a response on `return_chan`
+    pub fn send(&self, message: Message, return_chan: oneshot::Sender<Response>) -> Result<()> {
+        let message_id = if let Frame::Cassandra(frame) = &message.original {
+            frame.stream_id
+        } else {
+            info!("no cassandra frame found");
```
Comment (rukai) [inline shotover-proxy/src/transforms/cassandra/connection.rs:63]: I think returning Err should be enough, no need to also info log it.

**Comment by conorbros** [comment]
My PR on multi-map hasn't been merged yet so I haven't included that in this PR. I can point it at my fork or wait until it's merged and then switch over the `HashMap`?

**Comment by rukai** [comment]
Yeah HashMap can go in a follow up PR.
Keeps this PR more manageable too.

**Review by rukai — APPROVED** [review]
Some really nice improvements here!

**Review by benbromhead — APPROVED** [review]

**Pull request merged**